src/Components/PostComment.jsx

```diff
@@ -0,0 +1,43 @@
+import { useContext, useState } from "react"
+import { postCommentToArticle } from "../api"
+import { UserContext } from "../contexts/Users";
+
+function PostComment({setAllComments, article_id}){
+const [newComment, setNewComment] = useState('')
+const [isSubmitDisabled, setIsSubmitDisabled] = useState(false);
+
+const { user } = useContext(UserContext);
+
+const handleSubmit = (event) => {
+  event.preventDefault()
+setIsSubmitDisabled(true)
+  postCommentToArticle(article_id, newComment, user).then((newCommentFromApi) => {
+    console.log(newCommentFromApi)
+    setAllComments((currentComments) => {
+      return [newCommentFromApi, ...currentComments]
+    })
+    setNewComment('')
+    setIsSubmitDisabled(false)
+
+  })
+.catch((error) => {
+  const errorMessage = error.response.data.message || 'Something went wrong!';
+  alert(errorMessage);
+});
+}
+
+    return (
+        <form className="add-comment" onSubmit={handleSubmit}>
+        <label  htmlFor="new-comment"/>
+        <textarea
```

Comment:
Currently an empty comment can be posted - needs some logic to prevent that! Check out the 'required' property you can put on an input or textarea.

A bit of feedback in the case of a failed comment post would be great as well.